tests/components/downloader/test_config_flow.py

```diff
@@ -32,10 +31,7 @@ async def test_user_form(hass: HomeAssistant) -> None:
         )
         assert result["type"] is FlowResultType.FORM
 
-        with patch(
-            "homeassistant.components.downloader.config_flow.DownloaderConfigFlow._validate_input",
-            side_effect=DirectoryDoesNotExist,
-        ):
+        with patch("os.path.isdir", return_value=False):
```

Comment:
What is the point of this patch here?
It feels pointless as we only have asserts of the existing result here - no action.

Reply:
I'm not following the question.

We want to assert here that a missing directory shows an appropriate error in the form.

Reply:
What I mean is that I think this patch is in the wrong place.
If you change the return value to True, or change to raise an exception, the test still passes.
If you add `assert_any_call()` then it fails with `AssertionError: isdir() call not found`

I think maybe the original intention was to have it on line 27 instead of line 34.

Reply:
Right, there should be an additional configure call after this patch and we should outdent it. The patch on line 24 should be moved down above the last successful configure call.

Reply:
Good catch, thanks @epenet!